external_parser/example_joiner.cc

```diff
@@ -273,38 +288,66 @@ void example_joiner::try_set_label(const joined_event &je,
   examples[index]->l.cb.weight = weight;
 }
 
-int example_joiner::process_interaction(const v2::Event &event,
-                                        const v2::Metadata &metadata,
-                                        const TimePoint &enqueued_time_utc,
-                                        v_array<example *> &examples) {
+void example_joiner::clear_batch_info() {
+  _batch_grouped_events.clear();
+  _batch_grouped_examples.clear();
+  while (!_batch_event_order.empty()) {
+    _batch_event_order.pop();
+  }
+}
+
+void example_joiner::clear_event_id_batch_info(const std::string &id) {
+  _batch_grouped_events.erase(id);
+  _batch_event_order.pop();
+  _batch_grouped_examples.erase(id);
+}
+
+void example_joiner::invalidate_joined_event(const std::string &id) {
+  if (_batch_grouped_examples.find(id) != _batch_grouped_examples.end()) {
+    _batch_grouped_examples[id].ok = false;
+  }
+}
+
+bool example_joiner::process_interaction(const v2::Event &event,
+                                         const v2::Metadata &metadata,
+                                         const TimePoint &enqueued_time_utc,
+                                         v_array<example *> &examples) {
+
+  if (EnumNamePayloadType(metadata.payload_type()) !=
+      EnumNameProblemType(_problem_type_config)) {
+    VW::io::logger::log_warn(
```

Comment:
I find it weird that we're calling a global function and not an instance method

Reply:
I'll check with Peter but I think that that is how the logs are currently implemented, there is no instance method afaik

Reply:
There are plans to create an instanced logger, but it requires a lot of piping and since the change was kind of urgent I went with a global logger for now